drivers/clock_control/clock_control_esp32.c

```diff
@@ -110,12 +112,8 @@ static int clock_control_esp32_get_rate(const struct device *dev, clock_control_
 }
 
 #if defined(CONFIG_SOC_SERIES_ESP32)
-static void esp32_clock_perip_init(void)
+static bool reset_reason_is_cpu_reset(void)
```

Comment:
Having the single `reset_reason_is_cpu_reset` with multiple SOCs ifdefs seems to be more suitable to maintain the code readability.